kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/partition/Partitioner.java

```diff
@@ -18,16 +18,28 @@
  */
 
 
+import com.google.cloud.bigquery.TableId;
+
+import com.wepay.kafka.connect.bigquery.exception.BigQueryConnectException;
+import org.apache.kafka.connect.data.Schema;
+
 import java.util.List;
+import java.util.Set;
 
 /**
- * Interface for partitioning lists of elements.
+ * Interface for partitioning lists of elements and writing those partitions to bigQuery.
  * @param <E> The type of element in the list that will be partitioned.
  */
 public interface Partitioner<E> {
+
   /**
-   * @param elements The list of elements to partition.
-   * @return A series of lists, which when combined in order is equal to the argument list.
+   * @param table The BigQuery table to write the rows to.
+   * @param elements The list of elements to write to BigQuery.
+   * @param topic The Kafka topic that the row data came from.
+   * @param schemas The unique Schemas for the row data.
+   * @throws BigQueryConnectException if we are unable to write to BigQuery
+   * @throws InterruptedException if writing is interrupted
    */
-  public List<List<E>> partition(List<E> elements);
+  void writeAll(TableId table, List<E> elements, String topic, Set<Schema> schemas)
```

Comment:
I don't think that the `<E>` is needed anymore. Since the writer is actually used within the partitioner now, and the writer requires using `InsertAllRequest.RowToInsert`, I think it makes sense to just force `InsertAllRequest.RowToInsert` as the type.